src/LinearMaps.jl

```diff
@@ -18,8 +17,9 @@ using Base: require_one_based_indexing
 
 abstract type LinearMap{T} end
 
-const MapOrVecOrMat{T} = Union{LinearMap{T}, AbstractVecOrMat{T}}
-const MapOrMatrix{T} = Union{LinearMap{T}, AbstractMatrix{T}}
+const AbstractVecOrMatOrQ{T} = Union{AbstractVecOrMat{T}, AbstractQ{T}}
```

Comment:
I had never head of `AbstractQ` before so I searched it and it is a subtype of AbstractMatrix so why do we need this new union here?
https://github.com/JuliaLang/julia/blob/670190c3393e7a2193ad9e40cfa12ec643ceba75/stdlib/LinearAlgebra/src/qr.jl#L519

Reply:
Because of https://github.com/JuliaLang/julia/pull/46196. This is in preparation of that PR. It makes no difference on older Julia versions, but will on v1.10.

Reply:
I should perhaps split the two things in two separate PRs to not mix it up...

Reply:
Oh wow, I see.  Thanks for the explanation.  I am not looking forward to having to add more such `Union's` to support things that are kind of like a `AbstractMatrix` (in terms of multiplying) but not a subtype of it.  I've subscribed to the discussion of 46196 so I can see how that debate goes...  Now I'll try to see if this branch works for me.